File: sandbox-templates/performance-analyzer/java/BenchmarkRunner.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Random;

public class BenchmarkRunner {
    public static void main(String[] args) {
        int[] testSizes = {100, 1000, 5000};
        StringBuilder jsonResults = new StringBuilder("[");

        for (int i = 0; i < testSizes.length; i++) {
            int size = testSizes[i];
            ArrayList<Integer> testData = generateTestData(size);

            long startOriginal = System.nanoTime();
            OriginalCode.main(new ArrayList<>(testData));
            long endOriginal = System.nanoTime();

            long startImproved = System.nanoTime();
            ImprovedCode.main(new ArrayList<>(testData));
            long endImproved = System.nanoTime();

            double originalMs = (endOriginal - startOriginal) / 1_000_000.0;
            double improvedMs = (endImproved - startImproved) / 1_000_000.0;

            jsonResults.append(String.format(
                "{\\"size\\":%d,\\"original\\":%f,\\"improved\\":%f}",
                size, originalMs, improvedMs
            ));
            if (i < testSizes.length - 1) {
                jsonResults.append(",");
            }
        }
        jsonResults.append("]");
        System.out.println(String.format("{\\"results\\":%s,\\"unit\\":\\"ms\\"}", jsonResults.toString()));
    }

    private static ArrayList<Integer> generateTestData(int size) {
        ArrayList<Integer> list = new ArrayList<>(size);
        Random rand = new Random();
        for (int i = 0; i < size; i++) {
            list.add(rand.nextInt(size));
        }
        return list;
    }
}

File: sandbox-templates/performance-analyzer/js/benchmark_runner.mjs
const {main: originalFunction} = await import('./original_code.js');
const {main: improvedFunction} = await import('./improved_code.js');

const testSizes = [100, 1000, 5000];
const results = [];

function generateTestData(size) {
    return Array.from({length: size}, () => Math.floor(Math.random() * size));
}

for (const size of testSizes) {
    const testData = generateTestData(size);
    const startOriginal = performance.now();
    originalFunction([...testData]);
    const endOriginal = performance.now();

    const startImproved = performance.now();
    improvedFunction([...testData]);
    const endImproved = performance.now();

    results.push({size: size, original: endOriginal - startOriginal, improved: endImproved - startImproved});
}

console.log(JSON.stringify({results: results, unit: "ms"}));